libpod/in_memory_state.go

```diff
@@ -0,0 +1,279 @@
+package libpod
+
+import (
+	"github.com/docker/docker/pkg/truncindex"
+	"github.com/kubernetes-incubator/cri-o/pkg/registrar"
+	"github.com/pkg/errors"
+)
+
+// An InMemoryState is a purely in-memory state store
+type InMemoryState struct {
+	pods         map[string]*Pod
+	containers   map[string]*Container
+	podNameIndex *registrar.Registrar
+	podIDIndex   *truncindex.TruncIndex
+	ctrNameIndex *registrar.Registrar
+	ctrIDIndex   *truncindex.TruncIndex
+}
+
+// NewInMemoryState initializes a new, empty in-memory state
+func NewInMemoryState() (State, error) {
+	state := new(InMemoryState)
+
+	state.pods = make(map[string]*Pod)
+	state.containers = make(map[string]*Container)
+
+	state.podNameIndex = registrar.NewRegistrar()
+	state.ctrNameIndex = registrar.NewRegistrar()
+
+	state.podIDIndex = truncindex.NewTruncIndex([]string{})
+	state.ctrIDIndex = truncindex.NewTruncIndex([]string{})
+
+	return state, nil
+}
+
+// GetContainer retrieves a container from its full ID
+func (s *InMemoryState) GetContainer(id string) (*Container, error) {
```

Comment:
Can/should we drop Get, we are trying to avoid these, based on golang standards.

Reply:
Alright, renaming to Container and AllContainers for GetAllContainers. Will do same for pods.